File: from-2601-to-2700/2657-find-the-prefix-common-array-of-two-arrays/solution.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        counts = defaultdict(int)
        result = [0] * len(A)
        curr = 0

        for i, (a, b) in enumerate(zip(A, B)):
            counts[a] += 1
            if counts[a] == 2:
                curr += 1

            counts[b] += 1
            if counts[b] == 2:
                curr += 1
            
            result[i] = curr

        return result

File: from-2601-to-2700/2657-find-the-prefix-common-array-of-two-arrays/solution.go
func findThePrefixCommonArray(A []int, B []int) []int {
    common := 0
    n := len(A)
    result := make([]int, n)
    counts := make(map[int]int)
    for i := 0; i < n; i ++ {
        counts[A[i]] += 1
        if counts[A[i]] == 2 {
            common += 1
        }
        counts[B[i]] += 1
        if counts[B[i]] == 2 {
            common += 1
        }

        result[i] = common
    }
    return result
}